src/networking.c

```diff
@@ -378,6 +436,50 @@ void deleteCachedResponseClient(client *recording_client) {
 /* -----------------------------------------------------------------------------
  * Low level functions to add more data to output buffers.
  * -------------------------------------------------------------------------- */
+static inline void insertPayloadHeader(char *buf, size_t *bufpos, uint8_t type, size_t len, int slot, payloadHeader **new_header) {
```

Comment:
Since we call this function only once from `upsertPayloadHeader`, and it involves many complex parameters, consider removing it and placing the code directly in `upsertPayloadHeader`